### PR TITLE
Common/Network: Get rid of out parameters for MAC address utilities

### PR DESCRIPTION
**Pull request by lioncash**
Given we have std::array and std::optional, we can use these in conjunction with one another to avoid the need for out parameters.

### CODE REVIEW

File: Source/Core/Core/IOS/USB/Bluetooth/BTReal.cpp [hunk 1]
```diff
@@ -514,8 +514,7 @@ void BluetoothReal::LoadLinkKeys()
     if (index == std::string::npos)
       continue;
 
-    bdaddr_t address;
-    Common::StringToMacAddress(pair.substr(0, index), address.data());
+    bdaddr_t address = Common::StringToMacAddress(pair.substr(0, index)).value();
```
Comment (lioncash) [inline Source/Core/Core/IOS/USB/Bluetooth/BTReal.cpp]: @leoetlino just one thing I noticed while making changes, but is it safe to assume these addresses will always be valid like the previous code was doing? It doesn't seem like we sanitize the link keys anywhere beforehand to ensure they're valid MAC addresses before using them.

Comment (leoetlino) [reply Source/Core/Core/IOS/USB/Bluetooth/BTReal.cpp]: Hmm, good catch -- we should probably ignore the link key entry if the MAC address is invalid. (Though that can only normally happen if the user corrupted their own config file on purpose)

Comment (lioncash) [reply Source/Core/Core/IOS/USB/Bluetooth/BTReal.cpp]: Thanks, I added a commit to skip it if that situation ever happens; may as well protect users from themselves.

File: Source/Core/Core/IOS/Network/MACUtils.cpp [hunk 2]
```diff
@@ -30,19 +31,22 @@ void GetMACAddress(u8* mac)
...
+  if (!mac)
   {
-    Common::GenerateMacAddress(Common::MACConsumer::IOS, mac);
-    SaveMACAddress(mac);
+    mac = Common::GenerateMacAddress(Common::MACConsumer::IOS);
+    SaveMACAddress(mac.value());
     if (!wireless_mac.empty())
     {
       ERROR_LOG(IOS_NET,
                 "The MAC provided (%s) is invalid. We have "
```
Comment (riking) [inline Source/Core/Core/IOS/Network/MACUtils.cpp]: the error message acts like it's printing the invalid address, but it's actually printing the new generated address

Comment (lioncash) [reply Source/Core/Core/IOS/Network/MACUtils.cpp]: This isn't related to the changes being made.

**Pull request merged**